optaplanner-core/src/main/java/org/optaplanner/core/config/util/ConfigUtils.java

```diff
@@ -261,35 +263,31 @@ public static int ceilDivide(int dividend, int divisor) {
         return (dividend / divisor) + correction;
     }
 
-    /**
-     * Name of the variable that represents {@link Runtime#availableProcessors()}.
-     */
-    public static final String AVAILABLE_PROCESSOR_COUNT = "availableProcessorCount";
-
-    public static int resolveThreadPoolSizeScript(String propertyName, String script, String... magicValues) {
-        final String scriptLanguage = "JavaScript";
-        ScriptEngine scriptEngine = new ScriptEngineManager().getEngineByName(scriptLanguage);
-        if (scriptEngine == null) {
-            throw new IllegalStateException("The " + propertyName + " (" + script
-                    + ") could not resolve because the JVM doesn't support scriptLanguage (" + scriptLanguage + ").\n"
-                    + "Maybe try running in a normal JVM.");
+    public static int resolveThreadPoolSize(String propertyName, String script, IntSupplier defaultValue,
+            IntSupplier availableProcessors, Logger logger) {
+        int resolved;
+        if (script == null || script.equals(THREAD_COUNT_AUTO)) {
+            resolved = defaultValue.getAsInt();
+        } else {
+            try {
+                resolved = Integer.parseInt(script);
+            } catch (Exception e) {
+                throw new IllegalStateException("The " + propertyName + " (" + script +
+                        ") resolved to neither '" + THREAD_COUNT_AUTO + "' nor a number.");
```

Comment:
this is lacking UNLIMITED for one of the use cases. This is related to the removal of magicValues parameter.